src/Tasks.UnitTests/WriteLinesToFile_Tests.cs

```diff
@@ -0,0 +1,216 @@
+                File.Delete(file);
+            }
+        }
+
+        [Fact]
+        public void WriteLinesWriteOnlyWhenDifferentTest()
+        {
+            var file = FileUtilities.GetTemporaryFile();
+            try
+            {
+                // Write an initial file.
+                var a = new WriteLinesToFile
+                {
+                    Overwrite = true,
+                    BuildEngine = new MockEngine(),
+                    File = new TaskItem(file),
+                    WriteOnlyWhenDifferent = true,
+                    Lines = new ITaskItem[] { new TaskItem("File contents1") }
+                };
+
+                a.Execute().ShouldBeTrue();
+
+                // Verify contents
+                var r = new ReadLinesFromFile { File = new TaskItem(file) };
+                r.Execute().ShouldBeTrue();
+                r.Lines[0].ItemSpec.ShouldBe("File contents1");
+
+                var writeTime = DateTime.Now.AddHours(-1);
+
+                File.SetLastWriteTime(file, writeTime);
+
+                // Write the same contents to the file, timestamps should match.
+                var a2 = new WriteLinesToFile
+                {
+                    Overwrite = true,
+                    BuildEngine = new MockEngine(),
+                    File = new TaskItem(file),
+                    WriteOnlyWhenDifferent = true,
+                    Lines = new ITaskItem[] { new TaskItem("File contents1") }
+                };
+                a2.Execute().ShouldBeTrue();
+                File.GetLastWriteTime(file).ShouldBe(writeTime, tolerance: TimeSpan.FromSeconds(1));
+
+                // Write different contents to the file, last write time should differ.
+                var a3 = new WriteLinesToFile
+                {
+                    Overwrite = true,
+                    BuildEngine = new MockEngine(),
+                    File = new TaskItem(file),
+                    WriteOnlyWhenDifferent = true,
+                    Lines = new ITaskItem[] { new TaskItem("File contents2") }
+                };
+
+                a3.Execute().ShouldBeTrue();
+                File.GetLastWriteTime(file).ShouldBeGreaterThan(writeTime.AddSeconds(1));
+            }
+            finally
+            {
+                File.Delete(file);
+            }
+        }
+
+        /// <summary>
+        /// Should create directory structure when target <see cref="WriteLinesToFile.File"/> does not exist.
+        /// </summary>
+        [Fact(Skip = "skip")]
+        public void WriteLinesToFileDoesCreateDirectory()
+        {
+            using (var testEnv = TestEnvironment.Create())
+            {
+                var directory = testEnv.CreateFolder(folderPath: null, createFolder: false);
+                var file = Path.Combine(directory.Path, $"{Guid.NewGuid().ToString("N")}.tmp");
+
+                var WriteLinesToFile = new WriteLinesToFile
+                {
+                    BuildEngine = new MockEngine(),
```

Comment:
Can you pass a test logger to this engine for improved diagnosability when things go wrong? 123e49e9bb78559ee28c9ad87be477aaa214dbb3 is an example of the easy way to do this in this codebase--once you have `_output`, you can just pass it to the `MockEngine` constructor.

Reply:
Ok, got it - makes sense.